docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+Following application code will be used to explain APIs.
+
+     public class PureStyleSQLApplication implements StreamingApplication
+     {
+       @Override
+       public void populateDAG(DAG dag, Configuration conf)
+       {
+         SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+    
+         sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                        "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
+    
+         sqlEnv = sqlEnv.registerTable("SALES", new FileEndpoint("/tmp/output", 
+                                       "out.file", new CSVMessageFormat(<<csvSchema>>)));
+    
+         sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, 
+                                                                  "apex_concat_str");
+    
+         sqlEnv.executeSQL(dag, sql);
+       }// populateDAG finished
+       
+       public static String apex_concat_str(String s1, String s2)
+       {
+            return s1 + s2;
+        }
+     }
+      
+### Constructing SQLExecEnvironment
+We will go over APIs from above code one by one. Class _**SQLExecEnvironment**_ provides a starting point and simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates instance of SQLExecEnvironment.  
+
+    // Creates SQLExecEnvironment instance by using static method getEnvironment
+    SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+
+### Registering tables with SQLExecEnvironment
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+
+    // Register KafkaEnpoint as table with kafka topic as table and specify data format as CSV
+    sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                            "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
+    
+    // Register FileEndpoint as tbale with HDFS file as table and specify data format as csv
+    sqlEnv = sqlEnv.registerTable("SALES", 
+                                          new FileEndpoint("/tmp/output", "out.file", 
+                                          new CSVMessageFormat(schemaOutDef)));
+
+    
+**_"registerTable"_** API takes the name of the table and data store endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for data. Each of these endpoints requires a different type of configurations but all of them demand data format to be defined. In this example data format is defined using _**MessageFormat**_ by specifying it at endpoint creation. As of now 
+**_CSVMessageFormat_** is only supported **_Messageformat_** for _KafkaEndpoint_ and _FileEndpoint_. **_CSVMessageFormat_** can be configured with schema string as follows:
```

Comment:
MessageFormat are not specific to anything specific Endpoint... Its independent... Its incorrect to say its only for KafkaEndpoint & FileEndpoint.

Also have all in italic and non in bold. It looks bad.